core/spec/models/spree/variant_spec.rb

```diff
@@ -497,4 +497,17 @@
       expect(Spree::Variant.in_stock).to eq [in_stock_variant]
     end
   end
+
+  context '#volume' do
+    let(:variant_zero_width) { create(:variant, width: 0) }
+    let(:variant) { create(:variant) }
+
+    it 'it is zero if any dimension parameter is zero' do
+      expect(variant_zero_width.volume).to eq 0
+    end
+
+    it 'return the volume if the dimension parameters are different of zero' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.